From f2ab0ef682dafb509f7e906fc7def6a1e49e1fe9 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@freescale.com>
Date: Wed, 5 Aug 2015 20:05:34 +0300
Subject: [PATCH 235/452] dpaa2-eth: Fix FQ type in stats print

Fix a bug where the type of the Rx error queue was printed
incorrectly in the debugfs statistics

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@freescale.com>
[Xulin: Original patch taken from FSL LS2085 SDK EAR6.0,
LS2085A-SDK-SOURCE-20160304-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c |   16 +++++++++++++++-
 1 files changed, 15 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c
index 9683aa5..5d5fba1 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c
@@ -90,6 +90,20 @@ static const struct file_operations ldpaa_dbg_cpu_ops = {
 	.release = single_release,
 };
 
+static char *fq_type_to_str(struct ldpaa_eth_fq *fq)
+{
+	switch (fq->type) {
+	case LDPAA_RX_FQ:
+		return "Rx";
+	case LDPAA_TX_CONF_FQ:
+		return "Tx conf";
+	case LDPAA_RX_ERR_FQ:
+		return "Rx err";
+	default:
+		return "N/A";
+	}
+}
+
 static int ldpaa_dbg_fqs_show(struct seq_file *file, void *offset)
 {
 	struct ldpaa_eth_priv *priv = (struct ldpaa_eth_priv *)file->private;
@@ -110,7 +124,7 @@ static int ldpaa_dbg_fqs_show(struct seq_file *file, void *offset)
 		seq_printf(file, "%5d%16d%16s%16llu%16u\n",
 			   fq->fqid,
 			   fq->target_cpu,
-			   fq->type == LDPAA_RX_FQ ? "Rx" : "Tx conf",
+			   fq_type_to_str(fq),
 			   fq->stats.frames,
 			   fcnt);
 	}
-- 
1.7.5.4

